assets/src/edit-story/components/form/color/colorInput.js

```diff
@@ -0,0 +1,240 @@
+  top: 0;
+  left: 0;
+  border-radius: 50px;
+  width: 24px;
+  height: 24px;
+`;
+
+const buttonStyle = css`
+  overflow: hidden;
+  border: 1px solid;
+  border-color: ${({ theme }) => theme.colors.border.defaultNormal};
+  outline: none;
+  ${KEYBOARD_USER_SELECTOR} &:focus {
+    border-color: ${({ theme }) => theme.colors.border.defaultHover};
+    box-shadow: none !important;
+  }
+  background: transparent;
+`;
+
+const ColorButton = styled(Preview).attrs(buttonAttrs)`
+  border-radius: 4px;
+  ${buttonStyle}
+`;
+
+const ColorPreview = styled.div`
+  ${colorStyles}
+  top: 6px;
+  left: 6px;
+  padding: 0;
+  background: transparent;
+  cursor: pointer;
+`;
+
+const ColorPreviewButton = styled(ColorPreview).attrs(buttonAttrs)`
+  ${buttonStyle}
+  padding: 0;
+  border: none;
+`;
+
+const ColorPreviewInsideButton = styled(ColorPreview)`
+  border: none;
+  transform: translate(-1px, -1px);
+`;
+
+const CurrentColor = styled.div`
+  ${colorStyles}
+`;
+
+const Transparent = styled.div`
+  ${colorStyles}
+
+  background-image: conic-gradient(
+    #fff 0.25turn,
+    #d3d4d4 0turn 0.5turn,
+    #fff 0turn 0.75turn,
+    #d3d4d4 0turn 1turn
+  );
+  background-size: 66.67% 66.67%;
+`;
+
+const TextualPreview = styled.div`
+  padding: 6px 12px 6px 38px;
+  text-align: left;
+  flex-grow: 1;
+  height: 32px;
+`;
+
+function ColorInput({
+  onChange,
+  hasGradient,
+  hasOpacity,
+  value,
+  label,
+  colorPickerActions,
+  changedStyle,
+}) {
+  const isMixed = value === MULTIPLE_VALUE;
+  value = isMixed ? '' : value;
+
+  const previewStyle = getPreviewStyle(value);
+  const previewText = getPreviewText(value);
+
+  const [pickerOpen, setPickerOpen] = useState(false);
+  const previewRef = useRef(null);
+
+  const {
+    refs: { inspector },
+  } = useInspector();
+
+  const colorType = value?.type;
+  // Allow editing always in case of solid color of if color type is missing (mixed)
+  const isEditable = !colorType || colorType === 'solid';
+
+  const buttonProps = {
+    onClick: () => setPickerOpen(true),
+    'aria-label': label,
+  };
+
+  // Always hide color picker on unmount - note the double arrows
+  useUnmount(() => () => setPickerOpen(false));
+
+  const onClose = useCallback(() => setPickerOpen(false), []);
+  const spacing = useMemo(() => ({ x: 20 }), []);
+
+  return (
+    <>
+      {isEditable ? (
+        // If editable, only the visual preview component is a button
+        // And the text is an input field
+        <Preview ref={previewRef}>
+          <Input
+            aria-label={label}
+            value={value}
+            onChange={onChange}
+            placeholder={isMixed ? MULTIPLE_DISPLAY_VALUE : ''}
+          />
+          <ColorPreviewButton
+            {...buttonProps}
+            color={previewStyle?.backgroundColor}
+          >
+            {(value?.a < 1 || isMixed) && <Transparent />}
+            <CurrentColor role="status" style={previewStyle} />
+          </ColorPreviewButton>
```

Comment:
This button doesn't show focus in the tab order, so when you tab from the input field, you end up in a kind of no-mans-land until you tab again and get to the opacity field.

Reply:
Not actually sure what the focus display should be? Maybe just a blue circle around the color preview itself, if that can fit in? Also, should this button have a tooltip of "Open color picker"? And similarly for non-solid for the entire button?